File: frontend/src/constants/registration.ts
// 报名状态枚举（与后端 backend/internal/constants/registration.go 保持一致）
export const RegistrationStatus = {
  REGISTERED: 'registered',
  CANCELLED: 'cancelled',
  CHECKED_IN: 'checked_in',
} as const

export const RegistrationStatusText: Record<string, string> = {
  [RegistrationStatus.REGISTERED]: '已报名',
  [RegistrationStatus.CANCELLED]: '已取消',
  [RegistrationStatus.CHECKED_IN]: '已签到',
}

export const ReviewStatus = {
  PENDING: 'pending',
  APPROVED: 'approved',
  REJECTED: 'rejected',
} as const

export const ReviewStatusText: Record<string, string> = {
  [ReviewStatus.PENDING]: '待审核',
  [ReviewStatus.APPROVED]: '已通过',
  [ReviewStatus.REJECTED]: '已拒绝',
}

export const CheckInMethodText: Record<string, string> = {
  voucher: '凭证签到',
  scan: '扫码签到',
}

File: backend/internal/constants/registration.go
package constants

// RegistrationStatus 报名状态枚举。
const (
	RegistrationStatusRegistered = "registered"
	RegistrationStatusCancelled  = "cancelled"
	RegistrationStatusCheckedIn  = "checked_in"
)

// RegistrationStatusValues 全部报名状态值。
var RegistrationStatusValues = []string{RegistrationStatusRegistered, RegistrationStatusCancelled, RegistrationStatusCheckedIn}

// ReviewStatus 报名审核状态枚举。
const (
	ReviewStatusPending  = "pending"
	ReviewStatusApproved = "approved"
	ReviewStatusRejected = "rejected"
)

// ReviewStatusValues 全部审核状态值。
var ReviewStatusValues = []string{ReviewStatusPending, ReviewStatusApproved, ReviewStatusRejected}

// CheckInMethod 签到方式枚举。
const (
	CheckInMethodVoucher = "voucher"
	CheckInMethodScan    = "scan"
)

// IsValidRegistrationStatus 校验报名状态。
func IsValidRegistrationStatus(s string) bool {
	for _, v := range RegistrationStatusValues {
		if v == s {
			return true
		}
	}
	return false
}

// IsValidReviewStatus 校验审核状态。
func IsValidReviewStatus(s string) bool {
	for _, v := range ReviewStatusValues {
		if v == s {
			return true
		}
	}
	return false
}
